### ISSUE TITLE
[expo-router] Navigation to adjacent stacks nested in bottom tabs not behaving as expected (back button)

### ISSUE THREAD

**Issue by janusqa** [expo/expo#24691]
### Minimal reproducible example

https://github.com/janusqa/rn-bare/tree/bug_navigation

### Which package manager are you using? (Yarn is recommended)

npm

### If the issue is web-related, please select the bundler (`web.bundler` in the `app.json`)

None

### Summary

Expectation
If I navigate to a screen in an adjacent stack nested in a bottom tab
a) if the screen is the initial screen in that stack it should not have a back button
b) if the screen is not the initial screen in that stack it should always have a back button to the initial screen

Not sure if these are correct expectations, but it was what I was expecting.

However the behavior I get is that if I navigate to a screen in an adjacent stack and its not the initial screen in that stack I do not get a back button.  If i then click on the tab for that stack and go to the initial screen there is a back button.

To reproduce:
1) download and run repro repo
2)  It will load "Tab 1 / Screen 1"
3) click on the last link (Go to Tab 2 / Screen 2)
4) Observe that this screen has no back button (I think it should since its not the initial screen for the stack in this tab)
5) click on the second tab (to go to Tab 2 / Screen 1)
6) Observe that this screen has a back button (I think it should not since its the initial screen for this stack in this tab)

### Environment

```
expo-env-info 1.0.5 environment info:
    System:
      OS: Linux 5.4 Ubuntu 20.04.6 LTS (Focal Fossa)
      Shell: 5.0.17 - /bin/bash
    Binaries:
      Node: 18.14.0 - ~/.nvm/versions/node/v18.14.0/bin/node
      Yarn: 1.22.19 - ~/.nvm/versions/node/v18.14.0/bin/yarn
      npm: 9.4.2 - ~/.nvm/versions/node/v18.14.0/bin/npm
    SDKs:
      Android SDK:
        API Levels: 28, 29, 33, 34
        Build Tools: 30.0.3, 33.0.2, 34.0.0
        System Images: android-28 | Google Play Intel x86 Atom, android-29 | Google Play Intel x86 Atom, android-33 | Google APIs Intel x86_64 Atom, android-33 | Google Play Intel x86_64 Atom
    npmPackages:
      expo: ^49.0.8 => 49.0.8 
      react: 18.2.0 => 18.2.0 
      react-native: 0.72.4 => 0.72.4 
    Expo Workflow: managed
```

**Comment by JustinHaut** [comment]
this seems to be related to MaterialCommunityIcons, not expo-router. I'm using react-navigation and have the same issue with the nav icons as well as other icons. The issue is for Android only. Here's a past thread about it, but I haven't found a working solution yet.

https://github.com/expo/expo/issues/21568

**Comment by JJosephttg** [comment]
I also have this issue. The initialRouteName option does not help with this and I am not sure how to work around this except for providing your own back button which is not great

**Comment by Akumzy** [comment]
Hi @marklawlor, any update on this?
This is our biggest roadblock with `expo-router` and we have already done with migration from `react-navigation` before bumping into this.

**Comment by marklawlor** [comment]
This is being addressed in #28644
